corehq/apps/reports/filters/users.py

```diff
@@ -369,6 +363,14 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs, request_user=None):
             else:
                 return q.filter(id_filter)
 
+    @staticmethod
+    def _verify_users_are_accessible(domain, request_user, user_ids):
+        # This function would be very slow if called with many user ids
+        for user_id in user_ids:
+            other_user = CommCareUser.get(user_id)
+            if not user_can_access_other_user(domain, request_user, other_user):
+                raise PermissionDenied("One or more users are not accessible")
```

Comment:
`user_ids` is the list of users explicitly selected by the user, so I expect it'll generally be pretty small.